build.ps1

```diff
@@ -103,26 +103,25 @@ try {
 
     if ($run)
     {
-        # Run cleanup first so the samples don't get overloaded with old device instances
+        # Run cleanup first so the samples don't get overloaded with old devices
         RunApp provisioning\Samples\service\CleanupEnrollmentsSample "Provisioning\Service\CleanupEnrollmentsSample"
         RunApp iot-hub\Samples\service\CleanUpDevicesSample "IoTHub\Service\CleanUpDevicesSample"
 
-        RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample"
-        RunApp iot-hub\Samples\device\KeysRolloverSample "IoTHub\Device\KeysRolloverSample"
-        RunApp iot-hub\Samples\device\MessageSample "IoTHub\Device\MessageSample"
-        RunApp iot-hub\Samples\device\MethodSample "IoTHub\Device\MethodSample"
-        RunApp iot-hub\Samples\device\TwinSample "IoTHub\Device\TwinSample"
-
-        RunApp iot-hub\Samples\module\MessageSample "IoTHub\Module\MessageSample"
-        RunApp iot-hub\Samples\module\TwinSample "IoTHub\Module\TwinSample"
+        RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        RunApp iot-hub\Samples\device\MessageReceiveSample "IoTHub\Device\MessageReceiveSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        RunApp iot-hub\Samples\device\MethodSample "IoTHub\Device\MethodSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        RunApp iot-hub\Samples\device\TwinSample "IoTHub\Device\TwinSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"
+        # DeviceStreaming sample is not added since it is not available in all regions.
+        # DeviceReconnectionSample is not added since it is designed to run until cancelled using Console input.

```

Comment:
You could add a parameter to the sample for run duration too.

Reply:
Then the cancellationToken signaled by the control-C would conditionally have a timespan associated with it.

Reply:
I can give that a try, sure